dot/parachain/availability-distribution/availability_distribution_test.go

```diff
@@ -945,3 +948,144 @@ func TestAvailabilityChunkIndex(t *testing.T) {
 		})
 	}
 }
+
+func TestProcessAvailabilityDistributionMessageFetchPoV(t *testing.T) { //nolint:tparallel
+	t.Parallel()
+
+	var (
+		ctrl        *gomock.Controller
+		netMock     *MockNetwork
+		stateMock   *MockBlockState
+		runtimeMock *MockInstance
+		cacheMock   *MockSessionCache
+		overseerCh  chan any
+		ad          *AvailabilityDistribution
+
+		relayParent    = common.Hash{0x01}
+		validatorIndex = parachaintypes.ValidatorIndex(1)
+		authorityID    = parachaintypes.AuthorityDiscoveryID{0x05}
+
+		msg parachaintypes.AvailabilityDistributionMessageFetchPoV
+	)
+
+	setup := func(t *testing.T) {
+		ctrl = gomock.NewController(t)
+		defer ctrl.Finish()
+
+		netMock = NewMockNetwork(ctrl)
+		stateMock = NewMockBlockState(ctrl)
+		runtimeMock = NewMockInstance(ctrl)
+		cacheMock = NewMockSessionCache(ctrl)
+
+		netMock.EXPECT().RegisterRequestHandler(protocol.ID("req_chunk/2"), gomock.Any())
+		netMock.EXPECT().RegisterRequestHandler(protocol.ID("req_pov/1"), gomock.Any())
+
+		stateMock.EXPECT().GetRuntime(gomock.Any()).Return(runtimeMock, nil).AnyTimes()
+
+		cacheMock.EXPECT().
+			GetAuthorityID(
+				gomock.AssignableToTypeOf(validatorIndex),
+				gomock.AssignableToTypeOf(relayParent),
+				gomock.Any()).
+			Return(authorityID, nil).
+			AnyTimes()
+
+		overseerCh = make(chan any)
+		ad = NewAvailabilityDistribution(overseerCh, netMock, stateMock, cacheMock)
+		require.NotNil(t, ad)
+
+		msg = parachaintypes.AvailabilityDistributionMessageFetchPoV{
+			RelayParent:   relayParent,
+			FromValidator: validatorIndex,
+			PovCh:         make(chan parachaintypes.OverseerFuncRes[parachaintypes.PoV]),
+		}
+	}
+
+	testCases := []struct {
+		description         string
+		makeRequestResponse func() messages.ReqRespResult
+		errorExpected       bool
+	}{
+		{
+			description: "request_fails",
+			makeRequestResponse: func() messages.ReqRespResult {
+				return messages.ReqRespResult{
+					Response: nil,
+					Error:    errors.New("network failure"),
+				}
+			},
+			errorExpected: true,
+		},
+		{
+			description: "no_such_pov",
+			makeRequestResponse: func() messages.ReqRespResult {
+				response := messages.PoVFetchingResponse{}
+				require.NoError(t, response.SetValue(parachaintypes.NoSuchPoV{}))
+
+				return messages.ReqRespResult{
+					Response: &response,
+					Error:    nil,
+				}
+			},
+			errorExpected: true,
+		},
+		{
+			description: "happy_path",
+			makeRequestResponse: func() messages.ReqRespResult {
+				response := messages.PoVFetchingResponse{}
+				require.NoError(t, response.SetValue(parachaintypes.PoV{BlockData: []byte{0x01, 0x02}}))
+
+				return messages.ReqRespResult{
+					Response: &response,
+					Error:    nil,
+				}
+			},
+			errorExpected: false,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.description, func(t *testing.T) {
+			setup(t)
```

Comment:
Can we run them parallelly?

Reply:
No, because they share variables so that `setup()` doesn't need to have seven return values. I could change that, but all the tests in `dot/parachain/availability-distribution` take less than half a second to run on my laptop. If we want to speed up the test suite, we should probably look elsewhere first. :)

Reply:
Same as above: No because they share variables for mocks etc.